src/tree.rs

```diff
@@ -109,11 +112,11 @@ where
         &self,
         mut node_key: NodeKey,
         version: Version,
-        kvs: &[(KeyHash, OwnedValue)],
+        kvs: &[(KeyHash, Option<OwnedValue>)],
         depth: usize,
         hash_cache: &Option<&HashMap<NibblePath, [u8; 32]>>,
         tree_cache: &mut TreeCache<R>,
-    ) -> Result<(NodeKey, Node)> {
+    ) -> Result<Option<(NodeKey, Node)>> {
```

Comment:
Why does this return an `Option`?

Reply:
If the set of `kvs` caused deletions such that the entire subtree was deleted, there is no new node to be returned. The `Null` node is incorrect here, because, per comments in the codebase, the `Null` node is exclusively used to represent an empty root node:

https://github.com/penumbra-zone/jmt/blob/1bd64cbaaf44ddf52919b459e302b482ce5aed79/src/node_type.rs#L752-L753